Frameworks/CoreText/DWriteWrapper.mm

```diff
@@ -173,21 +178,74 @@ static CFStringRef _CFStringFromLocalizedString(IDWriteLocalizedStrings* localiz
 
     // TODO: #1001 attribs can be nil here?
     NSDictionary* attribs = [ts->_attributedString attributesAtIndex:range.location effectiveRange:NULL];
-    UIFont* font = [attribs objectForKey:static_cast<NSString*>(kCTFontAttributeName)];
+    CTFontRef font = static_cast<CTFontRef>([attribs objectForKey:static_cast<NSString*>(kCTFontAttributeName)]);
+
     CGFloat fontSize = kCTFontSystemFontSize;
+    std::vector<wchar_t> fontName(c_defaultFontName, c_defaultFontName + wcslen(c_defaultFontName));
+    DWRITE_FONT_WEIGHT weight = DWRITE_FONT_WEIGHT_REGULAR;
+    DWRITE_FONT_STYLE style = DWRITE_FONT_STYLE_NORMAL;
+    DWRITE_FONT_STRETCH stretch = DWRITE_FONT_STRETCH_NORMAL;
     if (font != nil) {
-        fontSize = [font pointSize];
+        fontSize = CTFontGetSize(font);
+        __InitDWriteFontPropertiesFromName(CTFontCopyName(font, kCTFontFullNameKey), &weight, &stretch, &style);
+        CFStringRef name = _DWriteGetFamilyNameForFontName(CTFontCopyName(font, kCTFontFullNameKey));
+        fontName.resize(CFStringGetLength(name) + 1);
+        CFStringGetCharacters(name, CFRangeMake(0, fontName.size()), reinterpret_cast<UniChar*>(fontName.data()));
     }
```

Comment:
Almost seems like a need for _DWriteGetFontFamilyNameAndProperties(...), so all of this can be done in one shot in multiple functions.  Right now, this segment is repeated.